internal/alias/target/alias.go

```diff
@@ -0,0 +1,92 @@
+// Copyright (c) HashiCorp, Inc.
+// SPDX-License-Identifier: BUSL-1.1
+
+package target
+
+import (
+	"context"
+
+	"github.com/hashicorp/boundary/internal/alias/target/store"
+	"github.com/hashicorp/boundary/internal/db/timestamp"
+	"github.com/hashicorp/boundary/internal/errors"
+	"github.com/hashicorp/boundary/internal/oplog"
+	"github.com/hashicorp/boundary/internal/types/resource"
+	"google.golang.org/protobuf/proto"
+)
+
+// A Alias contains a storage alias. It is owned by a scope.
+type Alias struct {
+	*store.Alias
+	tableName string `gorm:"-"`
+}
+
+func (al *Alias) clone() *Alias {
+	cp := proto.Clone(al.Alias)
+	return &Alias{
+		Alias: cp.(*store.Alias),
+	}
+}
+
+// allocAlias is just easier/better than leaking the underlying type
+// bits to the repo, since the repo needs to alloc this type quite often.
+func allocAlias() *Alias {
+	fresh := &Alias{
+		Alias: &store.Alias{},
+	}
+	return fresh
+}
+
+// NewAlias generates a new in-memory alias.
+func NewAlias(ctx context.Context, scopeId, v string, opt ...Option) (*Alias, error) {
+	const op = "alias.NewAlias"
```

Comment:
I think the ops here and elsewhere under alias/target should use target (repository_alias.go does)
```suggestion
	const op = "target.NewAlias"
```